soc/xtensa/intel_adsp/common/multiprocessing.c

```diff
@@ -62,8 +62,12 @@ bool soc_cpus_active[CONFIG_MP_NUM_CPUS];
  *
  * Note that alignment is absolutely required: the IDC protocol passes
  * only the upper 30 bits of the address to the second CPU.
+ *
+ * The .section is the same as __imr, but that can't be used for an assebly
+ * function like this
  */
-__asm__(".align 4                   \n\t"
+__asm__(".section .imr." Z_STRINGIFY(__FILE__) "." Z_STRINGIFY(__COUNTER__) "\n\t"
+	".align 4                   \n\t"
```

Comment:
The syntax here is fine, but the semantics are surprising.  What on earth is trying to launch a second core before the first has finished the IMR->HPSRAM copy?  That seems like the root cause here, nothing should be touching that code until long, long after we've exited early boot.

Reply:
Is this fixing the problem for you ? This code path is not exercised if you build with `CONFIG_MP_NUM_CPUS=1` but the boot still fails.

Reply:
> Is this fixing the problem for you ? This code path is not exercised if you build with `CONFIG_MP_NUM_CPUS=1` but the boot still fails.

Checked and like you said, whether it boots or not with `CONFIG_MP_NUM_CPUS=1`  doesn't depend on this code path. It does boot with the other change in `include/zephyr/arch/xtensa/cache.h`, though:
```
static ALWAYS_INLINE void z_xtensa_cache_flush(void *addr, size_t bytes)
```